ngraph/python/TUTORIAL.md

```diff
@@ -0,0 +1,216 @@
+
+The example above can be rewritten like this to use only operations from opset 4:
+
+```python
+import ngraph.opset4 as ops
+
+parameter_x = ops.parameter(shape=[2, 2], dtype=np.float32, name="x")
+parameter_b = ops.parameter(shape=[2, 2], dtype=np.float32, name="b")
+
+graph = ops.relu(parameter_x + parameter_b, name="y")
+```
+
+
+#### Creating an nGraph function from a graph
+
+You can combine a graph together with it's input parameters into an nGraph function object. An nGraph function can be passed to other parts of OpenVINO to perform analysis and inference.
+
+```python
+>>> function = ng.Function(graph, [parameter_x, parameter_b], "TestFunction")
+<Function: 'TestFunction' ({2,2})>
+```
+
+#### Running inference
+
+In order to run inference on an nGraph function, you'll need to convert it to an Inference Engine network
+
+```python
+>>> from openvino.inference_engine import IECore
+>>> 
+>>> ie_network = ng.function_to_cnn(function)
+>>> 
+>>> ie = IECore()
+>>> executable_network = ie.load_network(ie_network, 'CPU')
+>>> 
+>>> request = executable_network.requests[0]
+>>> request.infer({
+>>>    'x': np.array([[-1, 0], [1, 2]]),
+>>>    'b': np.array([[-1, 0], [0, 1]]),
+>>> })
+>>> request.output_blobs['y'].buffer
+array([[-0.,  0.],
+       [ 1.,  3.]], dtype=float32)
+```
+
+#### Loading an nGraph function from a file
+
+It's common to load a model from a file using the Inference Engine `read_network` method:
+
+    network = ie_core.read_network(model=model_path, weights=weights_path)
+
+An nGraph function can be extracted from an Inference Engine network using the `ngraph.function_from_cnn` method.
+The following example shows how an ONNX model can be loaded to an nGraph Function:
+
+```python
+from openvino.inference_engine import IECore
+ie = IECore()
+
+model_path='/path/to/model.onnx'
+network = ie.read_network(model=model_path)
+
+import ngraph as ng
+function = ng.function_from_cnn(network)
+```
+
+
+#### Inspecting an nGraph function
+
+You can use the nGraph function's `get_ordered_ops` method to get a topologically sorted list of it's graph `Node`s. 
+
+```python
+>>> function.get_ordered_ops()
+[<Parameter: 'b' ({2,2}, float)>,
+ <Parameter: 'x' ({2,2}, float)>,
+ <Add: 'Add_2' ({2,2})>,
+ <Relu: 'y' ({2,2})>,
+ <Result: 'Result_4' ({2,2})>]
+```
+
+Each `Node` has a unique `name` property, assigned at creation time. User provided names can be retrieved using the `get_friendly_name` method. `get_type_name` returns the operation type of the node and the `shape` property returns the shape of the node's output tensor.
+
+```python
+>>> for node in function.get_ordered_ops():
+>>>     print('Node name: {:15} Friendly name: {:10} Op: {:10} {}'.format(
+>>>         node.name, node.get_friendly_name(), node.get_type_name(), node.shape))
```

Comment:
Do we need to say smth about dynamic shapes here?

Reply:
Dynamic shapes are discussed further down.